src/utils/client-factory.ts

```diff
@@ -52,8 +55,35 @@ const clientFactory = <T extends ClientType>(
 
         case 'ApolloGraphQL':
             return new ApolloClient({
+                // https://www.apollographql.com/docs/react/performance/server-side-rendering#initializing-apollo-client
+                ssrMode: true, // prevents ApolloClient from refetching queries unnecessarily
                 cache: new InMemoryCache(),
                 uri,
+                headers,
+                link:
+                    // https://www.apollographql.com/docs/react/networking/advanced-http-networking/#overriding-options
+                    new HttpLink({
+                        uri,
+                        headers,
+                        fetchOptions: { method: 'GET' }, // override default POST to use GET
+                        // log uri (on dev mode) and fetch
+                        fetch: (...pl) => {
+                            if (process.env.NODE_ENV === 'production') {
+                                return fetch(...pl);
+                            }
+
+                            // https://github.com/apollographql/apollo-client/issues/4017
+                            // tweaked from musemind implementation
+
+                            // Uncomment below if we want to see outbound gql uri
+                            // const [uri] = pl;
+                            // console.log(uri)
+
```

Comment:
I commented out this part, but still wanted to kept this in case we need to log the URI for data mocking purposes